modules/calib3d/src/solvepnp.cpp

```diff
@@ -1009,7 +1009,7 @@ int solvePnPGeneric( InputArray _opoints, InputArray _ipoints,
 
     if (reprojectionError.needed())
     {
-        int type = reprojectionError.type();
+        int type = max(_ipoints.depth(), _opoints.depth()) == CV_64F ? CV_64F : CV_32F;
```

Comment:
This expression should be applied if:
- `.empty() == true`
- and `.fixedType() == false`

otherwise, use current `reprojectionError.type()`.

Reply:
Thanks @alalek, fixed it.